classes/submission/Submission.inc.php

```diff
@@ -248,6 +248,42 @@ function setHideAuthor($hideAuthor) {
 	}
 
 	/**
+	 * Set submission revision.
+	 * @param $submissionRevision int
+	 */
+	function setSubmissionRevision($submissionRevision) {
+		$this->setData('submissionRevision', $submissionRevision);
+	}
+
+	/**
+	 * Return submission revision.
+	 * @return int
+	 */
+	function getSubmissionRevision() {
+		return $this->getData('submissionRevision');
+	}
+
+	/**
+	 * Get the current revision id for a submission
+	 * @param $contextId int
+	 * @return int
+	 */
+	function getCurrentVersionId() {
+		$submissionId = $this->getId();
+		$contextId = $this->getContextId();
+
+		// handle publisher ids
+		if (!ctype_digit("$submissionId")) {
```

Comment:
hm, I don't know ho to debug this. The function getId() seems to be special for submissions, because of the custom ids ... any suggestions? 👼